### ISSUE TITLE
Release TDM pre-trained models on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [Luo-Yihong/TDM#1]
Hi @Luo-Yihong 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability.If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models for instance),
you can also claim the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I noticed that the README mentions you're planning to release pre-trained models. It'd be great to make them available on the 🤗 hub, to improve their discoverability/visibility.
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

See here for a guide: https://huggingface.co/docs/hub/models-uploading. 

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub. 

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels

**Comment by Luo-Yihong** [comment]
Hi @NielsRogge ,

Thanks for your interest in our work. We have released a bucket of TDM-LoRA. Please enjoy it!

- [TDM-SD3-LoRA](https://huggingface.co/Luo-Yihong/TDM_sd3_lora)
- [TDM-CogVideoX-2B-LoRA](https://huggingface.co/Luo-Yihong/TDM_CogVideoX-2B_LoRA)
- [TDM-Dreamshaper-LoRA](https://huggingface.co/Luo-Yihong/TDM_dreamshaper_LoRA)

Cheers,
Yihong

**Comment by NielsRogge** [comment]
Thanks, pinging @wauplin here regarding your [request](https://huggingface.co/Luo-Yihong/TDM_sd3_lora/discussions/1) to customize the code snippet when adding `library_name: diffusers` (I'm afraid that's not possible).

**Comment by Wauplin** [comment]
Hey there 👋 What would be the correct code snippets for those? For context, these code snippets are generated using [this code](https://github.com/huggingface/huggingface.js/blob/69ad498dbf1aa920a6297b072852715d2e426917/packages/tasks/src/model-libraries-snippets.ts#L266) based on the `diffusers` library tag + tags on the repo. It is possible to open a PR to update the snippet logic but first I'd like to know if it's not just a problem of tags (for instance tagging the repos as `lora` would add the "lora" tag).

**Comment by Wauplin** [comment]
tagging @sayakpaul here for viz' as diffusers expert
(I let you chose if it's worth opening a PR, I'm don't have much visibility/context on this)

**Comment by NielsRogge** [comment]
This one: https://huggingface.co/Luo-Yihong/TDM_sd3_lora?library=diffusers

**Comment by sayakpaul** [comment]
It does show the use of `load_lora_weights()` however, since we need to customize other stuff (like scheduler configs, LoRA scale, etc.), the default snippet won't be meaningful. 

However, this is almost a rarity (there's only a handful of LoRAs that enable few-steps inference). So, I think we don't need to change anything.
